FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+
+                if( xAckInfo.callback != NULL )
+                {
+                    xAckInfo.callback( xAckInfo.pCommandContext );
+                }
+
+                break;
+
+            case MQTT_PACKET_TYPE_PUBREC:
+            case MQTT_PACKET_TYPE_PUBREL:
+                break;
+
+            case MQTT_PACKET_TYPE_PINGRESP:
+
+                /* Nothing to be done from application as library handles
+                 * PINGRESP. */
+                LogWarn( ( "PINGRESP should not be handled by the application "
+                           "callback when using MQTT_ProcessLoop.\n\n" ) );
+                break;
+
+            /* Any other packet type is invalid. */
+            default:
+                LogError( ( "Unknown packet type received:(%02x).\n\n",
+                            pPacketInfo->type ) );
+        }
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvCommandLoop()
+{
+    Command_t xCommand;
+    Command_t xNewCommand;
+    Command_t * pxCommand;
+    MQTTStatus_t xStatus = MQTTSuccess;
+    static int lNumProcessed = 0;
+    bool xBreakOnNextProcessLoop = false;
+    bool xSubscribeProcessed = false;
+
+    while( 1 )
+    {
+        while( xQueueReceive( xCommandQueue, &xCommand, DEMO_TICKS_TO_WAIT ) )
+        {
+            pxCommand = &xCommand;
+
+            /* This demo requires the subscription to be present before the first publish. */
+            if( pxCommand->commandType == PUBLISH )
+            {
+                if( !xSubscribeProcessed )
+                {
+                    LogInfo( ( "Publish in queue before subscribe. Sending to back of queue." ) );
+                    prvAddCommandToQueue( pxCommand );
+                    continue;
+                }
+            }
+
+            xStatus = prvProcessCommand( pxCommand );
+
+            /* TODO: After reconnect implemented, add connect operation to front
+             * of queue if status was not successful. */
+            configASSERT( xStatus == MQTTSuccess );
+            lNumProcessed++;
+
+            if( pxCommand->commandType == PROCESSLOOP )
+            {
+                /* Add process loop back to end of queue. */
+                prvCreateCommand( PROCESSLOOP, NULL, NULL, &xNewCommand );
+                prvAddCommandToQueue( &xNewCommand );
+                lNumProcessed--;
+
+                if( xBreakOnNextProcessLoop )
+                {
+                    break;
+                }
+            }
+
+            /* Mark subscribed as being processed. */
+            if( pxCommand->commandType == SUBSCRIBE )
+            {
+                xSubscribeProcessed = true;
+            }
+
+            /* In this demo, exit after unsubscribing. */
+            if( pxCommand->commandType == UNSUBSCRIBE )
+            {
+                xBreakOnNextProcessLoop = true;
+            }
+        }
+
+        vTaskDelay( pdMS_TO_TICKS( 200 ) );
```

Comment:
I think defining a MACRO with this delay value would be good.

Reply:
Will remove this delay since it's not necessary